File: ISSUE-59/SOLUTION-1/get_endian.py
#!/usr/bin/env python
#-*- coding:utf-8 -*-
#
# 2014-06-27 by kantal59 
# License: LGPL
# OLIMEX WPC#59: Big – Little Endians
# 			Make code which detects on what machine is ran – with Big or Little Endian. 
# 			Can you do it in other language than C?

# I. The simple way
import sys
print( sys.byteorder)


# II: A more complicated way
import struct
s= struct.pack("=H",0x8000)	# '=': cpu native byte order but "standard" size; 	'H': unsigned short with 2 bytes (standard size)
if s[0]== '\00':
	print("Little endian")
else:
	print("Big endian")	

File: ISSUE-59/SOLUTION-1/get_endian.c
/*
 2014-06-27 by kantal59 
 License: LGPL
 OLIMEX WPC#59: Big – Little Endians
 			Make code which detects on what machine is ran – with Big or Little Endian. 
 			Can you do it in other language than C?
 			
 			Compile: gcc -o get_endian  get_endian
*/


#include <stdio.h>
#include <inttypes.h> 

int main(){

   uint16_t i = 0x8000;	
   char *p = (char *) &i;
   if (p[0] == 0) 
       puts( "LITTLE endian"); 	// 00 80 -> lowest address contains the least significant byte
   else
      puts( "BIG endian");	// smallest address contains the most significant byte
 
  return(0);
}

